app/routes/user/email_notifications.server.ts

```diff
@@ -0,0 +1,208 @@
+    })
+
+    // Materialized view
+    for (const topic of notification.topics) {
+      await db.email_notification_subscription.put({
+        uuid,
+        topic,
+        recipient: notification.recipient,
+      })
+    }
+  }
+
+  // Read
+  async getEmailNotifications() {
+    const db = await tables()
+    const results = await db.email_notification.query({
+      KeyConditionExpression: '#sub = :sub',
+      ExpressionAttributeNames: {
+        '#sub': 'sub',
+        '#uuid': 'uuid',
+        '#name': 'name',
+        '#created': 'created',
+        '#topics': 'topics',
+        '#recipient': 'recipient',
+      },
+      ExpressionAttributeValues: {
+        ':sub': this.#sub,
+      },
+      ProjectionExpression: '#uuid, #created, #name, #topics, #recipient',
+    })
+    const emailNotifications = results.Items as EmailNotificationVM[]
+    for (const notice of emailNotifications) {
+      let formats = []
+      notice.noticeTypes = []
+      for (const topic of notice.topics) {
+        let mappedData = mapTopicsToFormatAndNoticeType(topic)
+        formats.push(mappedData.noticeFormat)
+        notice.noticeTypes.push(mappedData.noticeType)
+      }
+      notice.format = [...new Set(formats)][0]
+    }
+    emailNotifications.sort((a, b) => a.created - b.created)
+    return emailNotifications
+  }
+
+  async getEmailNotification(uuid: string): Promise<EmailNotificationVM> {
+    const db = await tables()
+    const item = (await db.email_notification.get({
+      sub: this.#sub,
+      uuid,
+    })) as ({ sub: string } & EmailNotificationVM) | null
+    if (!item) throw new Response(null, { status: 404 })
+    let formats = []
+    item.noticeTypes = []
+    for (const topic of item.topics) {
+      let mappedData = mapTopicsToFormatAndNoticeType(topic)
+      formats.push(mappedData.noticeFormat)
+      item.noticeTypes.push(mappedData.noticeType)
+    }
+    item.format = [...new Set(formats)][0]
+    const { sub, ...notification } = item
+    return {
+      uuid,
+      ...notification,
+    }
+  }
+
+  // Update
+  async updateEmailNotification(email_notification: EmailNotification) {
+    if (!email_notification.uuid) return null
+    const db = await tables()
+    db.email_notification.update(
+      {
+        Key: { sub: this.#sub, uuid: email_notification.uuid },
+        UpdateExpression:
+          'set #name = :name, #recipient = :recipient, #topics = :topics, #active = :active ',
+        ExpressionAttributeNames: {
+          '#name': 'name',
+          '#topics': 'topics',
+          '#recipient': 'recipient',
+          '#active': 'active',
+        },
+        ExpressionAttributeValues: {
+          ':name': email_notification.name,
+          ':recipient': email_notification.recipient,
+          ':topics': email_notification.topics,
+          ':active': email_notification.active,
+        },
+      },
+      (err, data) => {
+        if (err) console.log(err)
+        else console.log(data)
+      }
```

Comment:
Use try/catch and async/await rather than error callbacks.

Reply:
I have removed this altogether, it was mainly just for debugging